form-view-model/src/main/kotlin/com/ritense/formviewmodel/viewmodel/ViewModelLoader.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.formviewmodel.viewmodel
+
+import com.ritense.valtimo.camunda.domain.CamundaTask
+import org.springframework.transaction.annotation.Transactional
+import kotlin.reflect.KClass
+
+@Transactional
+interface ViewModelLoader<T : ViewModel> {
+
+    fun load(task: CamundaTask? = null): T
+
+    fun supports(formName: String) = getFormName() == formName
+
+    @Suppress("UNCHECKED_CAST")
+    fun getViewModelType(): KClass<T> =
+        this::class.supertypes.first().arguments.first().type?.let { it.classifier as KClass<T> }
```

Comment:
This only works when `ViewModelLoader` is the first implemented interface, and is a direct implementation without inheritence.
```suggestion
        this::class.allSupertypes.first { it.classifier == ViewModelLoader::class }.arguments.first().type?.let { it.classifier as KClass<C> }
```